src/spell/modules/spell/spell/src/mailbox.rs

```diff
@@ -80,16 +62,27 @@ pub fn pop_mailbox() -> StringValue {
     }
 
     let db = db();
-    let result: eyre::Result<Option<String>> = try {
-        let mut get = db.prepare(r#" SELECT message, id FROM mailbox ORDER BY id DESC LIMIT 1"#)?;
-        let string = read_string(&mut get, 0)?;
-        let id = get.read::<i64>(1)?;
-
-        let mut delete = db.prepare(r#"DELETE FROM mailbox WHERE id = ?"#)?;
-        delete.bind(1, id)?;
-        delete.next()?;
-
-        string
+    let result: eyre::Result<Option<MailboxMessage>> = try {
+        let mut get = db.prepare(
+            r#" SELECT init_peer_id, timestamp, message, id FROM mailbox ORDER BY id DESC LIMIT 1"#,
+        )?;
+        let message = if let State::Row = get.next()? {
+            Some(MailboxMessage {
+                init_peer_id: get.read::<String>(0)?,
+                timestamp: get.read::<i64>(1)? as u64,
+                message: get.read::<String>(2)?,
+            })
+        } else {
+            None
+        };
+
+        if message.is_some() {
+            let id = get.read::<i64>(3)?;
```

Comment:
What will happen if `get.next()` isn't `State::Row` here? What will `get.read` exactly read?

Reply:
:see_no_evil: 
Maybe reorganize it somehow, then?
```rust
if message = if ... {
   // reads
   // delete
   Some(message)
} else {
  None
}
```

Reply:
Maybe, yeah. Firstly, copy-pasted from collection and then realized the problem and fixed it not soo pretty

Reply:
fixed